sdk/ml/azure-ai-ml/tests/internal_utils/unittests/test_ml_client.py

```diff
@@ -433,3 +433,33 @@ def test_ml_client_with_both_workspace_registry_names_throws(self, e2e_ws_scope:
             message
             == "Both workspace_name and registry_name cannot be used together, for the ml_client."
         )
+
+    def test_ml_client_with_cli_config(self, mock_credential):
+        kwargs = {
+            "cloud": "test_cloud",
+            "azure_portal": "https://test.portal.azure.com/",
+            "resource_manager": "https://test.management.azure.com/",
+            "active_directory": "https://test.login.microsoftonline.com/",
+            "aml_resource_id": "https://test.ml.azure.com/",
+            "storage_endpoint": "test.core.windows.net",
+            "registry_discovery_endpoint": "https://test.eastus.api.azureml.ms/"
+        }
+        ml_client = MLClient(
+                credential=mock_credential,
+                subscription_id=Test_Subscription,
+                resource_group_name=Test_Resource_Group,
+                workspace_name="test-ws1",
+                **kwargs,
+            )
+        assert ml_client._cloud == "test_cloud"
```

Comment:
can we verify that the endpoints are set properly here? Same below

Reply:
Best we can do is check the base_url, most of the cloud config is not saved in the client.  I'll add this to the test though.